scripts/init.sh

```diff
@@ -77,24 +77,26 @@ if [ -r $DEFAULT ]; then
     source $DEFAULT
 fi
 
-function pidofproc() {
-    if [ $# -ne 3 ]; then
-        echo "Expected three arguments, e.g. $0 -p pidfile daemon-name"
-    fi
+if ! typeset -F pidofproc &>/dev/null; then
+  function pidofproc() {
+      if [ $# -ne 3 ]; then
+          echo "Expected three arguments, e.g. $0 -p pidfile daemon-name"
+      fi
 
-    PID=`pgrep -f $3`
-    local PIDFILE=`cat $2`
+      pid=`pgrep -f '\'$3'\>'`
```

Comment:
I don't understand this pattern. Can you explain? Why isn't it something like?

`pgrep -f "$#$"` 

The last `$` anchors the process name.